action/protocol/staking/handler_candidate_selfstake.go

```diff
@@ -0,0 +1,95 @@
+package staking
+
+import (
+	"context"
+	"math"
+	"math/big"
+
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocol"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+)
+
+const (
+	handleCandidateActivate = "candidateActivate"
+
+	candidateNoSelfStakeBucketIndex = math.MaxUint64
+)
+
+func (p *Protocol) handleCandidateActivate(ctx context.Context, act *action.CandidateActivate, csm CandidateStateManager,
+) (*receiptLog, []*action.TransactionLog, error) {
+	actCtx := protocol.MustGetActionCtx(ctx)
+	featureCtx := protocol.MustGetFeatureCtx(ctx)
+	log := newReceiptLog(p.addr.String(), handleCandidateActivate, featureCtx.NewStakingReceiptFormat)
+	bucket, rErr := p.fetchBucket(csm, act.BucketID())
+	if rErr != nil {
+		return log, nil, rErr
+	}
+	// caller must be the owner of a candidate
+	cand := csm.GetByOwner(actCtx.Caller)
+	if cand == nil {
+		return log, nil, errCandNotExist
+	}
+
+	if err := p.validateBucketSelfStake(ctx, csm, NewEndorsementStateManager(csm.SM()), bucket, cand); err != nil {
+		return log, nil, err
+	}
+
+	log.AddTopics(byteutil.Uint64ToBytesBigEndian(bucket.Index), bucket.Candidate.Bytes())
+	// convert previous self-stake bucket to vote bucket
+	if cand.SelfStake.Cmp(big.NewInt(0)) > 0 {
```

Comment:
`if cand.SelfStake.Sign() > 0` is more efficient and clear

Reply:
good suggest